src/main/java/com/powsybl/openloadflow/sa/OpenSecurityAnalysis.java

```diff
@@ -271,21 +288,28 @@ private PostContingencyResult runPostContingencySimulation(LfNetwork network, Ac
         engine.getParameters().setVoltageInitializer(new PreviousValueVoltageInitializer());
         AcLoadFlowResult postContingencyLoadFlowResult = engine.run();
         boolean postContingencyComputationOk = postContingencyLoadFlowResult.getNewtonRaphsonStatus() == NewtonRaphsonStatus.CONVERGED;
-        List<LimitViolation> postContingencyLimitViolations = new ArrayList<>();
+        Map<Pair<String, Branch.Side>, LimitViolation> postContingencyLimitViolations = new HashMap<>();
         if (postContingencyComputationOk) {
             detectViolations(
                 network.getBranches().stream().filter(b -> !lfContingency.getBranches().contains(b)),
                 network.getBuses().stream().filter(b -> !lfContingency.getBuses().contains(b)),
                 postContingencyLimitViolations);
         }
 
+        preContingencyLimitViolations.forEach((subjectSideId, preContingencyViolation) -> {
+            LimitViolation postContingencyViolation = postContingencyLimitViolations.get(subjectSideId);
+            if (postContingencyViolation != null && postContingencyViolation.getAcceptableDuration() >= preContingencyViolation.getAcceptableDuration()) {
```

Comment:
@geofjamg I think that the comparator should rely on getValue and not on getAcceptableDuration. What do you think ?